src/sparseml/transformers/compression/README.md

```diff
@@ -30,52 +30,131 @@ needed for decompression in the compressed state_dict:
 }
 ```
 
+Config information gets stored in the HF config file
+```json
+// config.json
+{
+    "sparsity_config": {
+        "format": "sparse_bitmask", // "dense_sparsity" for original tensor format
+
+        // informational
+        "sparsity_structure": "unstructured", // or 2:4, 8:16 etc...
+        "global_sparsity": "0.5"
+    }
+}
+```
+
+## Saving/Loading Interface 
+
+Loading in a compressed model requires no interface changes
+
+```python
+from sparseml.transformers.utils import SparseAutoModelForCausalLM
+
+# should contain model.safetensors or model.safetensors.index.json
+model_path = "/PATH/TO/COMPRESSED_MODEL"
+
+model = SparseAutoModelForCausalLM.from_pretrained(
+    model_name_or_path=model_path,
+    **model_kwargs,
+)
+```
+
+Saving a compressed model with an explicitly provided compression config. The config
```

Comment:
Do we need this?

What is the advantage of enabling a user to provide a `SparsityConfig` as opposed to inferring it from the model attributes as per below?

This feels like a way that someone could trip themselves up

Reply:
This is mainly to future-proof, in the future we may have multiple types of sparse compression for instance. Currently, this allows the user to specify a DenseConfig() where we don't compress the model on disk but still store the global_sparsity and structure in the config

Reply:
That makes sense